app/actions/TrezorActions.js

```diff
@@ -421,6 +431,31 @@ const checkTrezorIsDcrwallet = () => async (dispatch, getState) => {
   if (addrValidResp.index !== 0) throw "Wallet replied with wrong index.";
 };
 
+// setStakeInputTypes adds a field to input that denotes stake spends. SSRTX
+function setStakeInputTypes(inputs, refTxs) {
+  const refs = {};
+  refTxs.forEach((ref) => (refs[ref.hash] = ref.bin_outputs));
+  for (let i = 0; i < inputs.length; i++) {
+    const input = inputs[i];
+    const bin_outputs = refs[input.prev_hash];
+    if (bin_outputs) {
```

Comment:
You can use the same early return technique here: `if (!bin_outputs) continue`, which further reduces indentation and cognitive load.

Reply:
Ok!

> cognitive load

memory usage?

Reply:
>> cognitive load
> memory usage?

No, cognitive load (i.e. "mental effort") while reading the code. For example:

I find it easier to reason about code with lower indentation levels and using the "early return" style. In this particular example: you can consider every `if (condition) continue` check as performing some specific validation and only entries that pass all validation checks run the actual useful code, which is located at the lowest indentation level of the loop.

I guess this is somewhat a subjective preference and I do know some people prefer it the other way around. Here's a more extensive write up on this style, with a lot of comments providing nuance on this style (https://blog.timoxley.com/post/47041269194/avoid-else-return-early).

Reply:
Thank you.